### PR TITLE
[New] Add wangEditor w/ git auto-update

### PR DESCRIPTION
**Pull request by sufuf3**
Pull request for issue: #
Related issue(s): #7268 #11170

Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required):https://github.com/wangfupeng1988/wangEditor
 * Official website (optional, not the repository):http://wangeditor.github.io/
 * NPM package url (optional):https://www.npmjs.com/package/wangeditor
 * License and its reference: MIT
 * GitHub / Bitbucket popularity (required):
   - Count of watchers: 122
   - Count of stars: 1272
   - Count of forks: 629
 * NPM download stats (optional):
   - Downloads in the last day: 71
   - Downloads in the last week: 639
   - Downloads in the last month: 1700

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [x] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

### CODE REVIEW

**Comment by sufuf3** [comment]
@cdnjs/intern3 
Please help me review this PR, Thank you.

File: ajax/libs/wangEditor/package.json [hunk 1]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "wangEditor",
+  "version": "2.1.23",
+  "filename": "js/wangEditor.min.js",
+  "description": "wangEditor - 基于javascript和css开发的 Web富文本编辑器 轻量简洁易用开源免费(wangEditor based on javascript and css that makes it lightweight, easy to create simple, free and opensource.)",
+  "homepage": "https://wangeditor.github.io/",
```
Comment (extend1994) [inline ajax/libs/wangEditor/package.json]: This redirect to http. So using http as the upstream would be better.

File: ajax/libs/wangEditor/package.json [hunk 2]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "wangEditor",
+  "version": "2.1.23",
+  "filename": "js/wangEditor.min.js",
+  "description": "wangEditor - 基于javascript和css开发的 Web富文本编辑器 轻量简洁易用开源免费(wangEditor based on javascript and css that makes it lightweight, easy to create simple, free and opensource.)",
```
Comment (extend1994) [inline ajax/libs/wangEditor/package.json]: I think we need the description only in English but not simply translation. Ref: https://github.com/cdnjs/cdnjs/pull/7268#r65663096, https://github.com/cdnjs/cdnjs/pull/10722#pullrequestreview-25726513,

File: ajax/libs/wangEditor/package.json [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+  "homepage": "https://wangeditor.github.io/",
+  "author": "wangfupeng <wangfupeng1988@163.com>",
+  "keywords": [
+    "wangEditor",
+    "editor",
+    "富文本编辑器"
```
Comment (extend1994) [inline ajax/libs/wangEditor/package.json]: Not sure if we should remove Chinese here...

File: ajax/libs/wangEditor/package.json [hunk 4]
```diff
@@ -0,0 +1,32 @@
+{
+  "name": "wangEditor",
+  "version": "2.1.23",
+  "filename": "js/wangEditor.min.js",
+  "description": "wangEditor - 基于javascript和css开发的 Web富文本编辑器 轻量简洁易用开源免费(wangEditor based on javascript and css that makes it lightweight, easy to create simple, free and opensource.)",
+  "homepage": "https://wangeditor.github.io/",
+  "author": "wangfupeng <wangfupeng1988@163.com>",
```
Comment (extend1994) [inline ajax/libs/wangEditor/package.json]: This is from bower.json. But I think we should follow package.json? And maybe you can confirm why `author` info is different between package.json and bower.json.

File: ajax/libs/wangEditor/package.json (hunk 2 above)
Comment (sufuf3) [inline ajax/libs/wangEditor/package.json]: @extend1994 I'll discuss with the author, about the English description.
If there is no response I'll try to use English to describe.

File: ajax/libs/wangEditor/package.json (hunk 1 above)
Comment (sufuf3) [inline ajax/libs/wangEditor/package.json]: The `https://wangeditor.github.io/` or `http://wangeditor.github.io/` will redirect to `http://www.wangeditor.com/`
`github.io` domain name support ssl, maybe we can use `https`.
What do you think?

File: ajax/libs/wangEditor/package.json (hunk 3 above)
Comment (sufuf3) [inline ajax/libs/wangEditor/package.json]: Neither do I. I was thought that google `富文本编辑器` will show the repo, so I follow the upstream to add it.

Comment (sufuf3) [inline ajax/libs/wangEditor/package.json]: @pvnr0082t could you help me about this keyword that should we take it?

File: ajax/libs/wangEditor/package.json (hunk 4 above)
Comment (sufuf3) [inline ajax/libs/wangEditor/package.json]: I was thought we just need the author name and mail info. and the GitHub link I think it's not necessary., so I follow the upstream's `bower.json`, not `package.json`.

If I modify as the followings will it better?
```js
  "author": {
    "name": "wangfupeng1988",
    "email": "wangfupeng1988@163.com",
    "url": "https://github.com/wangfupeng1988"
  },
```

File: ajax/libs/wangEditor/package.json (hunk 3 above)
Comment (pvnr0082t) [inline ajax/libs/wangEditor/package.json]: It will be better to use English keywords.

Comment (sufuf3) [inline ajax/libs/wangEditor/package.json]: OK I'll use English keywords.

**Comment by sufuf3** [comment]
@extend1994 I modified description in English, author, keywords.

But I didn't change homepage from `https` to `http`.
Because the `https://wangeditor.github.io/` or `http://wangeditor.github.io/` will redirect to `http://www.wangeditor.com/`
`github.io` domain name support ssl, I think we can use `https`.

Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
update to v3.0.0 due to the different file structure
ref: https://github.com/wangfupeng1988/wangEditor/issues/709#issuecomment-306083322
@cdnjs/intern3 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@cdnjs/library-reviewer 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@hare1039 @qwe19272375 @pvnr0082t @maruilian11 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@hare1039 @qwe19272375 @pvnr0082t @maruilian11
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@maruilian11 I add the `fonts` of 1.3.12.1~1.4.0
Please help me review this PR, Thank you.

File: ajax/libs/wangEditor/package.json [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+  "version": "3.0.0",
+  "filename": "wangEditor.min.js",
+  "description": "wangEditor, your simple open-source easy-use rich text editor!",
+  "homepage": "https://wangeditor.github.io/",
+  "author": {
+    "name": "wangfupeng1988",
+    "email": "wangfupeng1988@163.com",
+    "url": "https://github.com/wangfupeng1988"
+  },
+  "keywords": [
```
Comment (qwe19272375) [inline ajax/libs/wangEditor/package.json]: Should we add more keywords?

Comment (sufuf3) [inline ajax/libs/wangEditor/package.json]: So what keywords you suggest?
These keywords are good to google.
Upstream's keywords use Chinese, and we can't use it.

Comment (sufuf3) [inline ajax/libs/wangEditor/package.json]: @qwe19272375 could you suggest what keywords need to add?

Comment (qwe19272375) [inline ajax/libs/wangEditor/package.json]: Sorry, I have no clue too. I think these keywords are good enough.

**Comment by sufuf3** [comment]
@maruilian11 @pvnr0082t 
Please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@maruilian11 
Please help me review this PR, Thank you.

**Comment by maruilian11** [comment]
@PeterDaveHello The test seems to be stuck. Could you help deal with this? Thanks.

**Comment by sufuf3** [comment]
@dakshshah96 @cdnjs/intern3 @cdnjs/library-reviewer 
I modified the `homepage` field, please help me review this PR, Thank you.

**Comment by sufuf3** [comment]
@dakshshah96 @cdnjs/intern3 
Please help me review this PR again, Thank you.

**Comment by dakshshah96** [comment]
@sufuf3 LGTM.

Can't review because of "You can't perform that action at this time."

**Comment by PeterDaveHello** [comment]
I believe that the bug I mention here: https://twitter.com/PeterDaveHello/status/885966377093181441

**Pull request merged**